### ISSUE TITLE
ScriptBuilder generates different output with MSBuild vs. `dotnet sql-persistence`

### ISSUE THREAD

**Issue by agross** [Particular/NServiceBus.Persistence.Sql#847]
Hello,

first-timer SQL persistence user here.

I'm trying to wrap my head around the script generation. I maybe have a bit of a special case here because I like to keep my saga names short and use namespaces to separate them:

`./Features/Foo/Saga.cs`
`./Features/Bar/Saga.cs`

The `dotnet sql-persistence` script build does not like that because that would generate 2 e.g. `Saga_Create.sql` files. The MSBuild task doesn't care (no error during build).

I think the build should error out in the same way that `dotnet sql-persistence` does.

**Comment by DavidBoike** [comment]
Hey @agross,

Yeah, that's not how SQL Persistence is designed. The sagas will need to have independent class names - the namespace doesn't matter. This would also be an issue at the table level, because the tables the scripts create are going to be based on the class name as well.

I would recommend naming sagas something descriptive of what they're meant to control. Usually sagas define a set of business rules so they're usually some sort of `Policy` such as `ShipOrderPolicy` or `CustomerHasBecomePreferredPolicy` or `BuyersRemorsePolicy`, to use a few names from some of our canonical ecommerce-based examples.

**Comment by agross** [comment]
Yes, I've got around the name clash with the `SqlSaga` attribute. Still, I find it strange that the tool errors out while the msbuild integration does not.

**Comment by DavidBoike** [comment]
Yeah, that is strange. It would be interesting to know what exactly what is going on becuase I would prefer for MSBuild to fail as well.

So…

1. What is the error you're seeing with the command-line tool?
2. What does the solution look like? (Could you share it?)
3. What is the result of a "successful" build with MSBuild. What scripts do you get as a result? Does one saga's script overwrite the other?

**Comment by agross** [comment]
1. > What is the error you're seeing with the command-line tool?
    ```sh
    $ dotnet sql-persistence script build/bin/Feedback/Feedback.dll
    sql-persistence failed: File '/some/path/NServiceBus.Persistence.Sql/MsSqlServer/Sagas/Saga_Create.sql' already exists.
    ```
1. > What does the solution look like?

   Adding the `[SqlSaga(tableSuffix: "something")]` to both sagas named `<some namespace>.Saga`.
1. > What is the result of a "successful" build with MSBuild.

   The build directory contains a single `Saga_Create.sql` file. Given that the schema for both tables is shared, this is hard to detect. So yes, the first script was overwritten.

**Comment by DavidBoike** [comment]
So we've decided that this is a rough edge, but we don't have a timeline to fix it. I'm going to leave the issue open, and hopefully it can be picked up by a future release.
